### ISSUE TITLE
Add custom module block to YOLOv8

### ISSUE THREAD

**Issue by Jochiwon** [ultralytics/ultralytics#3564]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

hi, I'm trying to add custom module to YOLOv8.

I searched issues and follow several suggestions, but it didnt work.

**Things I tried**

1. add NewClass(nn.Module) to `nn/modules/block.py`
2. add 'NewClass' to `__all__` part in `block.py`
3. add NewClass to `from .block import ...` part in `nn/modules/ __init__.py `
4. add NewClass to `from ..../modules import ....` part in `nn/tasks.py`

but I continously get the same error.
![inv](https://github.com/ultralytics/ultralytics/assets/44675901/bc92558b-6d91-4d1e-bcfb-16d9cd44dafb)

of course NewClass is not part of `torch.nn`, the error might come from `globals()[m]` part.

and i think it means the class couldn't imported normally.

but what part I missing in Import part? 

I can't see any more part that related with import precedure.

What am i missing?? is there more file i have to change??

Please Help!!

Thanks.

p.s. without this error, my YOLOv8 code is good. I trained several baseline models and it worked normally.

**Comment by Laughing-q** [comment]
@Jochiwon can you also paste your custom yaml file here then I'm able to take a look?

**Comment by Kayzwer** [comment]
@Jochiwon have a check on the fucjtion `parse_model` in `task.py` maybe you need to add you model into the tuple(s).

**Comment by glenn-jocher** [comment]
@Kayzwer hi there! Thank you for reaching out and providing the details of the issue you're facing.

To add a custom module to YOLOv8, you'll need to make sure that your module is properly recognized and included in the model parsing process. One place to check is the `parse_model` function in `task.py`. In this function, you'll find tuples representing the supported YOLOv8 models. If your custom module is not included in one of these tuples, you'll need to add it there.

I hope this helps! If you have any further questions or need additional assistance, please feel free to ask.

**Comment by Jochiwon** [comment]
@Kayzwer ok. but i think .yaml is fine
![image](https://github.com/ultralytics/ultralytics/assets/44675901/98fe3237-b5cd-4edb-8a2e-d32e02d6763a)

**Comment by Jochiwon** [comment]
@Kayzwer @glenn-jocher 

are you guys mentioning "?" tuple part??
the error is ocurring at white line part.
of course i add NewClass to ? part.

![what](https://github.com/ultralytics/ultralytics/assets/44675901/205ce328-4992-4c4a-8773-2b5b7a283273)

**Comment by Jochiwon** [comment]
I uploaded few materials that might help.

<nn/modules/block.py>
![image](https://github.com/ultralytics/ultralytics/assets/44675901/55bc4cd6-8622-499e-9ec5-7cb9b0d3d5a5)


<nn/modules/__init__.py>
![image](https://github.com/ultralytics/ultralytics/assets/44675901/ac23fc01-92e6-4a92-ad84-9b9fbea24688)


<nn/tasks.py>
![image](https://github.com/ultralytics/ultralytics/assets/44675901/be29dad9-e446-499b-a52f-7f03260ef18f)

**Comment by glenn-jocher** [comment]
@Jochiwon hi there,

From the screenshots you provided, it seems like you have made the necessary changes in the `nn/modules/block.py`, `nn/modules/__init__.py`, and `nn/tasks.py` files to add your custom module (`NewClass`) to YOLOv8.

However, based on the error message you shared, it appears that the `globals()[m]` part is causing the issue. This suggests that the class is not being imported correctly.

To investigate further, you can check the following:

1. Make sure that the module containing `NewClass` is in the same directory as the other YOLOv8 modules, or provide the correct relative path when importing the module.
2. Verify that you have properly imported all the necessary modules in your custom module file (`NewClass`) and there are no missing dependencies.
3. Ensure that you have restarted your environment or reloaded the necessary modules after making the changes to the code.

In summary, double-check the import statements and file locations to ensure that your custom module is being imported correctly into YOLOv8.

If you still encounter issues, please provide more details about the structure of your custom module (`NewClass`) and any additional error messages or logs that could help us further diagnose the problem.

Hope this helps! Let us know if you have any other questions.

Best regards.

**Comment by moahaimen** [comment]
i have the same problem , i added a block to blocks.py and did all the steps and still same problem 

> I uploaded few materials that might help.
> 
> <nn/modules/block.py> ![image](https://user-images.githubusercontent.com/44675901/251970307-55bc4cd6-8622-499e-9ec5-7cb9b0d3d5a5.png)
> 
> <nn/modules/**init**.py> ![image](https://user-images.githubusercontent.com/44675901/251970333-ac23fc01-92e6-4a92-ad84-9b9fbea24688.png)
> 
> <nn/tasks.py> ![image](https://user-images.githubusercontent.com/44675901/251970417-be29dad9-e446-499b-a52f-7f03260ef18f.png)

i have same error i wish you have solved something, please if you have found anything tell me, thank you

**Comment by moahaimen** [comment]
> @Jochiwon hi there,
> 
> From the screenshots you provided, it seems like you have made the necessary changes in the `nn/modules/block.py`, `nn/modules/__init__.py`, and `nn/tasks.py` files to add your custom module (`NewClass`) to YOLOv8.
> 
> However, based on the error message you shared, it appears that the `globals()[m]` part is causing the issue. This suggests that the class is not being imported correctly.
> 
> To investigate further, you can check the following:
> 
> 1. Make sure that the module containing `NewClass` is in the same directory as the other YOLOv8 modules, or provide the correct relative path when importing the module.
> 2. Verify that you have properly imported all the necessary modules in your custom module file (`NewClass`) and there are no missing dependencies.
> 3. Ensure that you have restarted your environment or reloaded the necessary modules after making the changes to the code.
> 
> In summary, double-check the import statements and file locations to ensure that your custom module is being imported correctly into YOLOv8.
> 
> If you still encounter issues, please provide more details about the structure of your custom module (`NewClass`) and any additional error messages or logs that could help us further diagnose the problem.
> 
> Hope this helps! Let us know if you have any other questions.
> 
> Best regards.

@glenn-jocher dear sir, can you make a more detail explanation or screenshots so we can follow?

**Comment by Jochiwon** [comment]
@glenn-jocher @moahaimen Thx everyone. I solved the issue.
it is turned out that very funny mistakes! :)

I edited my local files, but in the code, it looks like the code import modules from `pip packages(ultralytics)`

so here's I tryed.

1. In the highest folder(ultralytics), I make one .py file. (ex. Train.py)
2. In the .py file, add `from ultralytics.yolo.engine.model import YOLO`
3. write command. (ex. model = YOLO(WEIGHT_PATH), model.train() ....)
4. now the code not use pip package file but local code!

**Comment by Kayzwer** [comment]
> @Kayzwer @glenn-jocher
> 
> are you guys mentioning "?" tuple part?? the error is ocurring at white line part. of course i add NewClass to ? part.
> 
> ![what](https://user-images.githubusercontent.com/44675901/251970180-205ce328-4992-4c4a-8773-2b5b7a283273.png)

`if m in (Classify, Conv`...
and
`if m in (BottleneckCSP`...

**Comment by moahaimen** [comment]
> @glenn-jocher @moahaimen Thx everyone. I solved the issue. it is turned out that very funny mistakes! :)
> 
> I edited my local files, but in the code, it looks like the code import modules from `pip packages(ultralytics)`
> 
> so here's I tryed.
> 
> 1. In the highest folder(ultralytics), I make one .py file. (ex. Train.py)
> 2. In the .py file, add `from ultralytics.yolo.engine.model import YOLO`
> 3. write command. (ex. model = YOLO(WEIGHT_PATH), model.train() ....)
> 4. now the code not use pip package file but local code!

Happy to you and thank you for the solution, but can give more explaination? Some screenshots please? Also is this oblnly what you didi and the blocks where auccessfully imported to the YOLOV8. YAML?

**Comment by glenn-jocher** [comment]
Dear @moahaimen,

Great to hear that you were able to solve the issue! The problem was due to the code importing modules from the pip package (ultralytics) instead of your local files. To overcome this, you followed these steps:

1. Created a new .py file in the highest folder (ultralytics).
2. Added the import statement `from ultralytics.yolo.engine.model import YOLO` to the .py file.
3. Implemented the necessary commands related to the YOLOv8 model, such as `model = YOLO(WEIGHT_PATH)` and `model.train()`.

By doing this, the code started using your local files instead of the pip package files. Unfortunately, we don't have access to the screenshots you mentioned, but following the steps you provided should ensure that the custom blocks are successfully imported into the YOLOv8 YAML.

If you need further assistance or have any additional questions, please don't hesitate to ask.

Best regards.

**Comment by moahaimen** [comment]
@glenn-jocher thank you i wish i understand well, these snapshots how i did your instructions i wish i am doing the instruction correctly
![image](https://github.com/ultralytics/ultralytics/assets/30658799/f3b17065-0019-4248-b243-88353e51feea)
![image](https://github.com/ultralytics/ultralytics/assets/30658799/ba4f69bd-64b5-4ff9-b71d-c303306381a9)
![image](https://github.com/ultralytics/ultralytics/assets/30658799/78a31c09-2939-4637-a7b5-4bebab2a969b)

**Comment by moahaimen** [comment]
@glenn-jocher  all above snapshots resulted to this  please help
![image](https://github.com/ultralytics/ultralytics/assets/30658799/bb680390-ec29-4ba2-bf44-ee433a0bdf1a)

**Comment by moahaimen** [comment]
@glenn-jocher @Jochiwon   i did all your instructions as in these snapshots  i wish my work is corrections
![image](https://github.com/ultralytics/ultralytics/assets/30658799/89686219-7492-41d6-babf-7cc4bbdf82c2)
then i went to/content/yolov8/ultralytics/nn/modules/block.py
![image](https://github.com/ultralytics/ultralytics/assets/30658799/0ec75ae7-b697-4bc8-939f-18f97be07eeb)
![image](https://github.com/ultralytics/ultralytics/assets/30658799/3d550ef0-afac-4e5b-8d90-fd001e991056)

then in /content/yolov8/ultralytics/nn/modules/__init__.py
![image](https://github.com/ultralytics/ultralytics/assets/30658799/c09e9559-c419-4010-9a8f-123d58018f85)

then /content/yolov8/ultralytics/nn/tasks.py
![image](https://github.com/ultralytics/ultralytics/assets/30658799/284b7f22-76fe-421a-afdf-7e43f7d25a24)
![image](https://github.com/ultralytics/ultralytics/assets/30658799/44c00e4d-cb9e-42d5-a9fb-d57eaa16f7ac)
![image](https://github.com/ultralytics/ultralytics/assets/30658799/63f2cd80-a445-4113-9f60-cf9f2e34b84d)
![image](https://github.com/ultralytics/ultralytics/assets/30658799/79b8c43e-0140-4fe2-ad5f-68b8a31d802a)
then i Yolov8.yaml
![image](https://github.com/ultralytics/ultralytics/assets/30658799/c3a79f61-52e6-40cf-88f8-657cf2e9a801)
and in Google Colab i used the CLI
!yolo task=detect mode=train model=/content/yolov8/ultralytics/models/v8/yolov8.yaml data=/content/datasets/dws/data.yaml epochs=15

and this error appeared 
![image](https://github.com/ultralytics/ultralytics/assets/30658799/e03d0535-01b7-4dab-a639-72c132fffc36)

@glenn-jocher @Jochiwon  i really wish you could help me 
thank you

**Comment by glenn-jocher** [comment]
@moahaimen hello,

Thank you for providing the snapshots and detailed steps you followed to add your custom code to YOLOv8. From the screenshots, it appears that you have made the necessary modifications to the `block.py`, `__init__.py`, `tasks.py`, and `Yolov8.yaml` files.

Regarding the error you encountered, the traceback suggests that there might be an issue with the definition of the `NewClass` or possibly a problem with the import statements in the modified files. I would recommend double-checking the following:

1. Verify that the `NewClass` is correctly defined in the `block.py` file and that there are no syntax errors or missing dependencies.
2. Ensure that the import statements in `__init__.py` and `tasks.py` point to the correct location of the `NewClass` module.
3. Check that the modifications made in `Yolov8.yaml` to include the `NewClass` are accurate and in the correct format.

Additionally, it would be helpful to provide the full error message and traceback to get a better understanding of the issue.

Feel free to reach out if you have any further questions or require additional assistance. We're here to help!

Best regards.

**Comment by moahaimen** [comment]
@glenn-jocher  ok sir the code in this file is correct?
![image](https://github.com/ultralytics/ultralytics/assets/30658799/cab745a2-e2f4-4242-95d6-12f433f1300f)

**Comment by glenn-jocher** [comment]
@moahaimen hi there,

Regarding the code in the given screenshot, it appears to be correct. The weight path should be specified in the `Yolov8.yaml` file rather than in the code directly.

To clarify, the weight path should be set in the `weights` field within the `Yolov8.yaml` file. You can provide the path to your trained YOLOv8 weights in this field.

Please make sure to update the `weights` field in the `Yolov8.yaml` file with the correct weight path before running the code.

If you encounter any further issues or have additional questions, please let us know. We're here to assist you.

Best regards.

**Comment by moahaimen** [comment]
@glenn-jocher  sir i dint find  a weight field in the Yolov8.yaml
![image](https://github.com/ultralytics/ultralytics/assets/30658799/4d231607-4bc1-46ae-8e43-e35ea3a0e221)
****

**Comment by glenn-jocher** [comment]
@Jochiwon hi there,

Thank you for reaching out. In YOLOv8, the weight parameter is specified differently compared to YOLOv5. Instead of using the `weights` field in the `Yolov8.yaml` file, you would need to pass the weight path as a command-line argument when running the code.

For example, you can modify your command to include the weight path:
```
!yolo task=detect mode=train model=/content/yolov8/ultralytics/models/v8/yolov8.yaml data=/content/datasets/dws/data.yaml epochs=15 weights=/path/to/weights.pt
```
Replace `/path/to/weights.pt` with the actual path to your trained YOLOv8 weights.

Make sure to update your command accordingly and provide the correct weight path in the command-line argument.

If you have any further questions or need additional assistance, feel free to ask. We're here to help!

Best regards.

**Comment by moahaimen** [comment]
> command

@glenn-jocher  Thank you sir for responding, but  if i understand you response right, first i didn't train any pt files i am just prectising if the block working so i can re-program it, so why i am using weights to make sure my block working i didnt understand that i need just to check it if its working first,

**Comment by glenn-jocher** [comment]
Dear @moahaimen,

Thank you for reaching out and for your question regarding the usage of weights in YOLOv8. Please allow me to clarify.

In YOLOv8, the weights parameter is typically used when you want to evaluate or use the pre-trained YOLOv8 model, which requires loading the weights file generated from the training process. However, if you are just practicing and testing the functionality of your custom block, you don't necessarily need to use weights.

To make sure your block is working correctly, you can follow these steps:
1. Verify that your custom block is defined properly and there are no syntax errors or missing dependencies.
2. Test the block in isolation by creating a simple script or notebook where you can instantiate the block and validate its functionality with sample inputs.

By taking the above steps, you can ensure that your custom block is working as intended without the need for training weights.

Please let me know if you have any further questions or need any additional clarification. I'd be happy to assist you further.

Best regards.

**Comment by moahaimen** [comment]
@glenn-jocher  the block works when testing it, but the problem that i cant call in the Yolov8.yaml,
i am trying to train my Yolov8.yaml using this block so i can save a pre-trained pt file, but i need to check the performance first and i did all the steps you told me sir and still have no idea whats the problem,

**Comment by moahaimen** [comment]
> @glenn-jocher @moahaimen Thx everyone. I solved the issue. it is turned out that very funny mistakes! :)
> 
> I edited my local files, but in the code, it looks like the code import modules from `pip packages(ultralytics)`
> 
> so here's I tryed.
> 
> 1. In the highest folder(ultralytics), I make one .py file. (ex. Train.py)
> 2. In the .py file, add `from ultralytics.yolo.engine.model import YOLO`
> 3. write command. (ex. model = YOLO(WEIGHT_PATH), model.train() ....)
> 4. now the code not use pip package file but local code!

@Jochiwon i really really need your help, please just check my snapshots

**Comment by glenn-jocher** [comment]
@moahaimen hi there,

Glad to hear that you were able to solve the issue! It seems that the problem was related to the import statements and module locations.

Based on your explanation, it appears that you were modifying the local files in the ultralytics repository, but the code was still importing modules from the installed pip package. To resolve this, you created a separate Python file in the highest folder of the ultralytics repository (let's call it `Train.py`). In `Train.py`, you imported the required module (`YOLO`) from the local code (ultralytics.yolo.engine.model).

By doing this, you were able to use the modified code from the local repository instead of the installed pip package when running the `Train.py` script.

I hope this explanation clarifies the steps you took to resolve your issue. If you have any further questions or need additional assistance, please don't hesitate to ask.

Best regards.

**Comment by moahaimen** [comment]
@glenn-jocher Yes thank you very much for your help, you helped me a lot in several issues i really do appreciate that to you, sir but i suffer now from several problems
1- when writing the code  "please Note that i chose the Yolov8.yaml"
![image](https://github.com/ultralytics/ultralytics/assets/30658799/8f01e106-687e-4118-8a29-b4941a254016)
there is no detection or Recall or precision or mAP, also finally says the result of the bus   say there is no detection 
![image](https://github.com/ultralytics/ultralytics/assets/30658799/78b8ece4-b98d-4615-9117-7e9eb8b56c3b)

2- when i change to Yolov8n.pt  the code works well and gives me the detection needed
![image](https://github.com/ultralytics/ultralytics/assets/30658799/d4b039af-4501-4ded-a71f-d7fbe33c0392)

can you help me with finding why that's happening

**Comment by glenn-jocher** [comment]
Dear @moahaimen,

Thank you for sharing your concerns and explaining the issues you are facing with YOLOv8. I'm here to help you with troubleshooting these problems.

Regarding the first issue, where the YOLOv8.yaml configuration file does not show metrics like detection, recall, precision, and mAP, and reports that there is no detection, it could be related to different factors. Please make sure to check the following:

1. Verify that your YOLOv8.yaml file is configured correctly. Ensure that the necessary parameters, such as classes, anchors, and strides, are properly set according to your specific dataset and requirements.

2. Double-check the annotation format of your dataset. YOLOv8 expects the annotations to be in a specific format, such as Darknet or COCO format. Make sure that your dataset annotations follow the correct format.

3. Confirm that your dataset has a sufficient number of labeled objects for detection. If the objects in your images are not labeled or are labeled incorrectly, it can result in the "no detection" output.

Regarding the second issue, where using the Yolov8n.pt model produces the desired detection results, it indicates that the issue may not be related to the code itself but rather to the specific weights file you are using. It's possible that the Yolov8.pt model you mentioned might not be trained on the same dataset or may have different anchor settings.

I recommend double-checking the training process and ensuring that you are using the correct weights file that matches your dataset and model configuration.

If you have any further questions or need additional assistance in resolving these issues, please let me know. I'm here to assist you.

Best regards.

**Comment by moahaimen** [comment]
@glenn-jocher  thank you for answering sir, i used the CoCo dataset and the Default Yolov8.yaml , without any changes on it 
and i dont think the problem related with dataset becuase when i use the same dataset with pretrained model Yolov8n.pt  it works well, 
 the problem is the training doesnt happen when choosing The Yolov8.yaml as it read but doesnt apply

**Comment by moahaimen** [comment]
@glenn-jocher  also, I think I found the problem it seems about the initializing weights, it's too low so 3 epochs don't give any detection there must be over 30 till training reach a suitable point, which is related to the Yolov8.yaml i don't know how and why the initialization weights changed, could please tell me why and how to reset it to the default because i went to the yolo>cfg> defaults its all normal why its changing  when i train using the Yaml

i think the problem is here 
----------------------------------------------------------------------------------------------------
optimizer: AdamW(lr=0.000119, momentum=0.9) with parameter groups 57 weight(decay=0.0), 64 weight(decay=0.0005), 63 bias(decay=0.0)
Image sizes 640 train, 640 val
-----------------------------------------------------------------------------------------------------------
 @glenn-jocher

**Comment by moahaimen** [comment]
> @moahaimen hello,
> 
> Thank you for providing the snapshots and detailed steps you followed to add your custom code to YOLOv8. From the screenshots, it appears that you have made the necessary modifications to the `block.py`, `__init__.py`, `tasks.py`, and `Yolov8.yaml` files.
> 
> Regarding the error you encountered, the traceback suggests that there might be an issue with the definition of the `NewClass` or possibly a problem with the import statements in the modified files. I would recommend double-checking the following:
> 
> 1. Verify that the `NewClass` is correctly defined in the `block.py` file and that there are no syntax errors or missing dependencies.
> 2. Ensure that the import statements in `__init__.py` and `tasks.py` point to the correct location of the `NewClass` module.
> 3. Check that the modifications made in `Yolov8.yaml` to include the `NewClass` are accurate and in the correct format.
> 
> Additionally, it would be helpful to provide the full error message and traceback to get a better understanding of the issue.
> 
> Feel free to reach out if you have any further questions or require additional assistance. We're here to help!
> 
> Best regards.

this solution worked for my laptop but it doesn work for Google Colab ,  i created the file and wrote the code in it , and i excute it by this  !python /content/yolov8/ultralytics/developing_blocks.py
but the same error of MDC  "    m = getattr(torch.nn, m[3:]) if 'nn.' in m else globals()[m]  # get module
KeyError: 'MDC'"

**Comment by glenn-jocher** [comment]
@moahaimen Hello,

Thank you for providing additional details regarding the error you encountered when running your modified YOLOv8 code in Google Colab.

Based on the error message you shared, it seems that there is an issue with the "MDC" module. The error suggests that the code is trying to access the "MDC" module but is unable to find it or it is not imported correctly.

To troubleshoot this issue, please consider the following steps:

1. Double-check that you have imported all the required modules correctly in your code. Make sure that the "MDC" module is properly imported or defined before using it.

2. Verify that the version of the YOLOv8 code and the required dependencies are compatible. It's possible that there may be differences in how the code is structured or version-dependent changes between your local environment and Google Colab.

3. Ensure that you have installed all the necessary packages and dependencies in your Google Colab environment. You can use the `!pip install` command to install missing packages directly in the Colab notebook.

If the issue persists, it would be helpful to provide more information, such as the specific code snippet that triggers the error and the full error message with the complete traceback. This will help us further investigate and identify the root cause of the problem.

Please let me know if you have any additional questions or need further assistance. We are here to help you!

Best regards.

**Comment by moahaimen** [comment]
@glenn-jocher 
1- I DID THAT
2-I UPDATED YOLOV8 USING !pip3 install --upgrade ultralytics
3- all the necessary packages and dependencies in the Google Colab environment have been installed 
the problem still
as I told you, sir, the process works fine on my laptop but its too old and has a weak GPU, so I really need to use Google Colab, I wish to get your help

**Comment by glenn-jocher** [comment]
@moahaimen hi,

I understand that you're facing an issue with running YOLOv8 in Google Colab. Despite performing the necessary steps such as importing the required modules, updating YOLOv8 using the command `!pip3 install --upgrade ultralytics`, and ensuring all necessary packages and dependencies are installed in your Colab environment, the problem persists.

Given that the same code works without any issues on your laptop (despite the weak GPU), it suggests that there might be some compatibility or configuration issue specific to the Google Colab environment.

To further troubleshoot the problem, here are a couple of recommendations:

1. Verify that you have correctly imported all the required modules and defined any custom classes or functions used in your code. Check for any potential syntax errors or missing dependencies that might be causing the specific error you encountered.

2. Validate that you have the necessary GPU requirements and configurations in your Google Colab environment. Ensure that you have selected a GPU runtime in Colab by going to "Runtime" > "Change runtime type" and selecting "GPU." This step is crucial to leverage the GPU capabilities for faster processing.

If the issue persists after these steps, please provide more details such as the specific error message with the complete traceback. This additional information will help us better understand the problem and provide you with further assistance.

Please let me know if there's anything else I can help you with. We're here to support you.

Best regards.

**Comment by moahaimen** [comment]
@glenn-jocher solved with simple solution only  moving the developing_blocks.py file from ultralytics for higher level folder , thank you  very much for your help sir

**Comment by moahaimen** [comment]
@glenn-jocher  i replaced the C2F block with the MDC block, and in training, it appeared
![image](https://github.com/ultralytics/ultralytics/assets/30658799/49379084-d7a7-4b02-aa59-59bbb1bee216)
but the performance is still the same, while the MDC is very different  from C2F, can you help me in that

**Comment by glenn-jocher** [comment]
@moahaimen hi there,

Thank you for sharing your experience with replacing the C2F block with the MDC block in YOLOv8 training. It's great to see that the MDC block is appearing during training.

Regarding the performance, it's important to note that the MDC block is a different architecture from the C2F block. While the MDC block may have different characteristics and potential benefits, its impact on performance can vary depending on several factors such as the dataset, training setup, and specific task requirements.

To further analyze the performance of the MDC block in YOLOv8, you can consider the following steps:

1. Evaluate the detection metrics: Check the precision, recall, and mAP (mean average precision) values to assess the performance of the model using the MDC block. These metrics provide insights into the model's ability to accurately detect objects in the dataset.

2. Compare the detected objects: Compare the objects detected using the MDC block with the objects detected using the original C2F block. Assess if there are any noticeable differences in terms of accuracy, object detection boundaries, or any other relevant factors.

3. Experiment with different hyperparameters: Try adjusting the hyperparameters specific to the MDC block, such as the number of channels or the configuration of the sub-modules, to see if they have any impact on the performance. Optimizing these hyperparameters can potentially improve the results.

Remember, the performance of the MDC block can vary based on the specific use case, dataset, and other factors. It may require additional experimentation and parameter tuning to achieve optimal results for your particular application.

If you need further assistance or have any specific questions or concerns regarding the implementation of the MDC block in YOLOv8, feel free to ask. We're here to help!

Best regards.

**Comment by moahaimen** [comment]
@glenn-jocher  sir the problem is not from my side, yolov8 works well if i dont add a block, but when i add a block as we did earlier and it worked well, before after today update of yolov8 now importing the blocks would make  all these problems i think now the architecture of yolov8 is against importing new block, so if you can please help me with this issue thank you also if out friends @Jochiwon @Kayzwer @Laughing-q has similar problem i wish they can help me fixing it ,
